src/main/resources/assets/react/document/useDocumentsState.ts

```diff
@@ -0,0 +1,353 @@
+	COLUMN_NAME_DOCUMENT_TYPE,
+	COLUMN_NAME_LANGUAGE,
+	COLUMN_NAME_ID,
+	COLUMN_NAME_JSON,
+];
+
+const OPTIONS_COLUMNS_DEFAULT: DropdownItemProps[] = [{
+	text: 'Collection',
+	value: COLUMN_NAME_COLLECTION, // 'document_metadata.collection'
+},{
+	text: 'Document type',
+	value: COLUMN_NAME_DOCUMENT_TYPE, // 'document_metadata.documentType'
+},{
+	text: 'Language',
+	value: COLUMN_NAME_LANGUAGE, // 'document_metadata.language'
+},{
+	text: 'Document ID',
+	value: COLUMN_NAME_ID
+},{
+	text: 'JSON',
+	value: COLUMN_NAME_JSON
+}]
+
+export function useDocumentsState({
+	servicesBaseUrl
+} :{
+	servicesBaseUrl: string
+}) {
+	// console.debug('servicesBaseUrl', servicesBaseUrl);
+
+	//──────────────────────────────────────────────────────────────────────────
+	// State
+	//──────────────────────────────────────────────────────────────────────────
+	const [documentsRes, setDocumentsRes] = React.useState({
+		total: 0,
+		hits: []
+	});
+
+	const [jsonModalState, setJsonModalState] = React.useState<{
+		open: boolean
+		id: string
+		parsedJson: Record<string,unknown>
+	}>({
+		open: false,
+		id: '',
+		parsedJson: undefined,
+	});
+
+	const [collectionOptions, setCollectionOptions] = React.useState<
+		DropdownItemProps[]
+	>([]);
+	const [collectionsHoverOpen, setCollectionsHoverOpen] = React.useState(false);
+
+	const [columnOptions, setColumnOptions] = React.useState<
+		DropdownItemProps[]
+	>(OPTIONS_COLUMNS_DEFAULT);
+
+
+	const [columnsHoverOpen, setColumnsHoverOpen] = React.useState(false);
+
+	const [documentTypeOptions, setDocumentTypeOptions] = React.useState<
+		DropdownItemProps[]
+	>([]);
+
+	const [documentTypesHoverOpen, setDocumentTypesHoverOpen] = React.useState(false);
+
+	const [operatorState, setOperatorState] = React.useState<DslOperator>(QUERY_OPERATOR_OR);
+	const [query, setQuery] = React.useState<string>('');
+
+	const [selectedCollections, setSelectedCollections] = React.useState<string[]>([]);
+	const [selectedColumns, setSelectedColumns] = React.useState(SELECTED_COLUMNS_DEFAULT);
+	const [selectedDocumentTypes, setSelectedDocumentTypes] = React.useState<string[]>([]);
+
+	//──────────────────────────────────────────────────────────────────────────
+	// Callbacks
+	//──────────────────────────────────────────────────────────────────────────
+	const queryDocuments = React.useCallback(({
+		collectionsFilter = [],
+		documentsTypesFilter = [],
+		operator = QUERY_OPERATOR_OR,
+		query = '',
+		selectedColumns = SELECTED_COLUMNS_DEFAULT,
+		updateSelectedCollections = false,
+		updateSelectedDocumentTypes = false,
+	}: {
+		collectionsFilter?: string[]
+		documentsTypesFilter?: string[]
+		operator?: DslOperator
+		query?: string
+		selectedColumns?: string[]
+		updateSelectedCollections?: boolean
+		updateSelectedDocumentTypes?: boolean
+	}) => {
+		// console.debug('queryDocuments');
+		const filters = [];
+		if (collectionsFilter.length) {
+			filters.push({
+				hasValue: {
+					field: 'document_metadata.collection',
```

Comment:
This should be moved to the const section at the top and perhaps combined with `FIELD_PATH_META`.